sonar/modules/documents/dojson/contrib/marc21tojson/model.py

```diff
@@ -474,299 +474,140 @@ def marc21_to_abstract(self, key, value):
-                                value,
-                                'a',
-                                default='').strip()
-    if subfield_a:
-        identifier['value'] = subfield_a
-        if value.get('q'):  # $q is repetitive
-            identifier['qualifier'] = \
-                ', '.join(utils.force_list(value.get('q')))
-        subfield_b = not_repetitive(marc21tojson.bib_id,
-                                    key,
-                                    value,
-                                    'b',
-                                    default='').strip()
-        if subfield_b:
-            identifier['source'] = subfield_b
-        # key[3] is the indicateur_1
-        identifier['type'] = type_for_ind1.get(key[3], 'bf:Identifier')
-        identifiedBy = self.get('identifiedBy', [])
-        identifiedBy.append(identifier)
-    return identifiedBy or None
+    return identified_by
 
 
 @marc21tojson.over('identifiedBy', '^035..')
 @utils.ignore_value
-def marc21_to_identifiedBy_from_field_035(self, key, value):
+def marc21_to_identified_by_from_035(self, key, value):
     """Get identifier from field 035."""
-    subfield_a = not_repetitive(marc21tojson.bib_id,
-                                key,
-                                value,
-                                'a',
-                                default='').strip()
-    if subfield_a:
-        identifier = {
-            'value': subfield_a,
-            'type': 'bf:Local',
-            'source': 'RERO'
-        }
-        identifiedBy = self.get('identifiedBy', [])
-        identifiedBy.append(identifier)
-    return identifiedBy or None
+    identified_by = self.get('identifiedBy', [])
+
+    if not value.get('a'):
+        return None
+
+    identified_by.append({
+        'type': 'bf:Local',
+        'source': 'RERO',
+        'value': value.get('a')
+    })
 
+    return identified_by
 
-@marc21tojson.over('identifiedBy', '^930..')
+
+@marc21tojson.over('identifiedBy', '^037..')
 @utils.ignore_value
-def marc21_to_identifiedBy_from_field_930(self, key, value):
-    """Get identifier from field 930."""
-    subfield_a = not_repetitive(marc21tojson.bib_id,
-                                key,
-                                value,
-                                'a',
-                                default='').strip()
-    if subfield_a:
-        identifier = {}
-        match = re.search(r'^\((.+?)\)\s*(.*)$', subfield_a)
-        if match:
-            # match.group(1) : parentheses content
-            identifier['source'] = match.group(1)
-            # value without parenthesis and parentheses content
-            identifier['value'] = match.group(2)
-        else:
-            identifier['value'] = subfield_a
-        identifier['type'] = 'bf:Local'
-        identifiedBy = self.get('identifiedBy', [])
-        identifiedBy.append(identifier)
-    return identifiedBy or None
+def marc21_to_identified_by_from_037(self, key, value):
```

Comment:
You should be sure that identifiedBy is in only one of the 037, 008, 091, else you will lose some informations.